packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -185,53 +191,101 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
 
   @override
   void paint(Canvas canvas, Size size) {
-    final Paint paint = Paint()
-      ..color = backgroundColor
-      ..style = PaintingStyle.fill;
+    final double effectiveTrackGap = switch (value) {
+      null || 1.0 => 0.0,
+      _ => trackGap ?? 0.0,
+    };
+
+    final Rect trackRect;
+    if (value != null && effectiveTrackGap > 0) {
+      trackRect = switch (textDirection) {
+        TextDirection.ltr => Rect.fromLTRB(
+          clampDouble(value!, 0.0, 1.0) * size.width + effectiveTrackGap,
+          0,
+          size.width,
+          size.height,
+        ),
+        TextDirection.rtl => Rect.fromLTRB(
+          0,
+          0,
+          size.width - clampDouble(value!, 0.0, 1.0) * size.width - effectiveTrackGap,
+          size.height,
+        ),
+      };
+    } else {
+      trackRect = Offset.zero & size;
+    }
+
+    // Draw the track.
+    final Paint trackPaint = Paint()..color = trackColor;
+    if (indicatorBorderRadius != null) {
+      final RRect trackRRect = indicatorBorderRadius!.resolve(textDirection).toRRect(trackRect);
+      canvas.drawRRect(trackRRect, trackPaint);
+    } else {
+      canvas.drawRect(trackRect, trackPaint);
+    }
 
-    paint.color = valueColor;
+    void drawStopIndicator() {
+      // Limit the stop indicator radius to the height of the indicator.
+      final double radius = math.min(stopIndicatorRadius!, size.height / 2);
+      final Paint indicatorPaint = Paint()..color = stopIndicatorColor!;
+      final Offset position = switch (textDirection) {
+        TextDirection.rtl => Offset(size.height / 2, size.height / 2),
+        TextDirection.ltr => Offset(size.width - size.height / 2, size.height / 2),
+      };
+      canvas.drawCircle(position, radius, indicatorPaint);
+    }
+
+    // Draw the stop indicator.
+    if (value != null && stopIndicatorRadius != null && stopIndicatorRadius !> 0) {
```

Comment:
Should this be:
```suggestion
    if (value != null && stopIndicatorRadius != null && stopIndicatorRadius! > 0) {
```
?

Reply:
Great catch!